### PR TITLE
recreate select2 on filter

### PR DESCRIPTION
**Pull request by calellowitz**
@orangejenny 
I suspect there is a cleaner way to do this but passing in the jquery element for the modal was the simplest way I could find. Happy to incorporate any changes to improve this.

JIRA Ticket:https://dimagi-dev.atlassian.net/browse/HI-545

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]
I wish there weren't a flag being passed around, but I checked this out and wasn't able to modify it in a way that removed the flag and still worked.

Passing around the modal is fine - I don't think that's avoidable without getting into using custom knockout bindings or using a knockout component for the overall data corrections UI, which is way out of scope.

File: corehq/apps/reports/static/reports/js/data_corrections.js [hunk 1]
```diff
@@ -101,9 +101,17 @@ hqDefine("reports/js/data_corrections", [
...
-        assertProperties.assert(options, ['saveUrl', 'properties'],
-            ['propertyNames', 'propertyNamesUrl', 'displayProperties', 'propertyPrefix', 'propertySuffix', 'analyticsDescriptor']);
+        assertProperties.assert(options, ['saveUrl', 'properties'],[
+            'propertyNames',
+            'propertyNamesUrl',
+            'displayProperties',
+            'propertyPrefix',
+            'propertySuffix',
+            'analyticsDescriptor',
+            '$modal'
```
Comment (orangejenny) [inline corehq/apps/reports/static/reports/js/data_corrections.js]: Missed this earlier - shouldn't `$modal` be a required property not an optional one?

Comment (calellowitz) [reply corehq/apps/reports/static/reports/js/data_corrections.js]: fixed

**Review by orangejenny — APPROVED** [review]
Switched github label because this has user-visible effects.

**Pull request merged**